source/hash_table.c

```diff
@@ -173,37 +164,22 @@ static int s_update_template_size(struct hash_table_state *template, size_t expe
         min_size = 2;
     }
 
-    size_t mask = ~(size_t)0, size = 1;
-    while (size < min_size) {
-        size = size << 1;
-        mask = mask << 1;
-
-        if (size == 0) {
-            /* Overflow */
-            return aws_raise_error(AWS_ERROR_OOM);
-        }
-    }
-    mask = ~mask;
-
-    /* Cross-check - make sure we didn't just overflow somehow. */
-    if (size < expected_elements) {
-        return aws_raise_error(AWS_ERROR_OOM);
+    /* size is always a power of 2 */
+    size_t size;
+    if (aws_round_up_to_power_of_two(min_size, &size)) {
+        return AWS_OP_ERR;
     }
 
+    /* Don't update the template until we've calculated everything successfully */
```

Comment:
This comment is somewhat weird: you're doing a lot of updates to the template below, so which calculation are you referring to?

Reply:
Clarified the comment.